onnx/backend/test/cmd_tools.py

```diff
@@ -86,11 +86,14 @@ def prepare_dir(path: str) -> None:
                             assert case.model.graph.input[j].type.HasField(
                                 "tensor_type"
                             )
-                            f.write(
-                                numpy_helper.from_array(
-                                    input, case.model.graph.input[j].name
-                                ).SerializeToString()
-                            )
+                            if isinstance(input, TensorProto):
```

Comment:
I guess this is because numpy doesn't natively support 8-bit floats? I wonder if we need something similar to what is done for bfloat16 (use some 8-bit view type from numpy) for inter-operation with numpy?

Reply:
True. Numpy does not support blfoat16 or float8. You can see how i modified the reference implementation to use custom dtype. I tried to  make it simple and intuitive. Feel free to tell it is not. Finding the most simple way to implement things is not always easy.